src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -111,397 +111,166 @@ MeshLevel::MeshLevel( string const & name,
-    SortedArray< localIndex > & allNodes  = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "all" ) ).reference();
-    allNodes.reserve( m_nodeManager->size() );
-
-    for( localIndex a=0; a<m_nodeManager->size(); ++a )
-    {
-      allNodes.insert( a );
-    }
-
+    m_edgeManager->localToGlobalMap()[iter_localToGlobalsize] = source.m_edgeManager->localToGlobalMap()[iter_localToGlobalsize];
   }
+  m_edgeManager->constructGlobalToLocalMap();
+  m_edgeManager->nodeList().resize( numLocalEdges, numNodesPerEdge );
 
 
+  /////////////////////////
+  // Faces
+  //////////////////////////
+
+  m_faceManager->resize( numLocalFaces );
+  m_faceManager->faceCenter() = source.m_faceManager->faceCenter();
+  m_faceManager->faceNormal() = source.m_faceManager->faceNormal();
 
-  ArrayOfArraysView< localIndex const > const & faceToNodeMapSource = source.m_faceManager->nodeList().toViewConst();
+  // copy the faces-to-edgs map from source
+  m_faceManager->edgeList() = source.m_faceManager->edgeList();
+  // copy the faces-to-elements map from source
+  m_faceManager->elementList() = source.m_faceManager->elementList();
+  m_faceManager->elementRegionList() = source.m_faceManager->elementRegionList();
+  m_faceManager->elementSubRegionList() = source.m_faceManager->elementSubRegionList();
+  // copy the faces-boundaryIndicator from source
+  m_faceManager->getDomainBoundaryIndicator() = source.m_faceManager->getDomainBoundaryIndicator();
 
-  FaceManager::ElemMapType const & faceToElem = source.m_faceManager->toElementRelation();
+  arrayView1d< globalIndex const > const faceLocalToGlobal = m_faceManager->localToGlobalMap();
+  for( localIndex iter_localToGlobalsize = 0; iter_localToGlobalsize < numLocalFaces; iter_localToGlobalsize++ )
+  {
+    m_faceManager->localToGlobalMap()[iter_localToGlobalsize] = source.m_faceManager->localToGlobalMap()[iter_localToGlobalsize];
+  }
+  m_faceManager->constructGlobalToLocalMap();
 
-  arrayView2d< localIndex const > const & faceToElemIndex = faceToElem.m_toElementIndex.toViewConst();
 
+  /////////////////////////
+  // Elements
+  //////////////////////////
 
+  // check that all elements are hexahedra
   source.m_elementManager->forElementRegions< CellElementRegion >( [&]( CellElementRegion const & sourceRegion )
   {
+    // create element region with the same name as source element region "Region"
     CellElementRegion & region = *(dynamic_cast< CellElementRegion * >( m_elementManager->createChild( sourceRegion.getCatalogName(),
                                                                                                        sourceRegion.getName() ) ) );
-
+    // add cell block to the new element region with the same name as cell block name from source element region
     region.addCellBlockNames( sourceRegion.getCellBlockNames() );
 
     sourceRegion.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion const & sourceSubRegion )
```

Comment:
For this precise copying loop, could you investigate if this would make sense to move then content into a `CellElementSubRegion::copyFromCellElementSubRegion` like we already have a `CellElementSubRegion::copyFromCellBlock` which is similar?
And maybe factorize some code in it if it makes sense?

I am not 100% sure of it, but surely it's worth having a look.